packages/itwin/quantity-formatting/src/components/quantityformat/internal/ThousandsSeparator.tsx

```diff
@@ -0,0 +1,171 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import * as React from "react";
+import type { FormatProps } from "@itwin/core-quantity";
+import { Format, FormatTraits, getTraitString } from "@itwin/core-quantity";
+import { Checkbox, IconButton, Label } from "@itwin/itwinui-react";
+import { SvgHelpCircularHollow } from "@itwin/itwinui-icons-react";
+import { useTranslation } from "../../../useTranslation.js";
+import { ThousandsSelector } from "./misc/ThousandsSelector.js";
+
+/** Properties of [[UseThousandsSeparator]] component.
+ * @internal
+ */
+export interface UseThousandsSeparatorProps {
+  formatProps: FormatProps;
+  onChange?: (format: FormatProps) => void;
+}
+
+/** Component to enable/disable the use of thousand separator.
+ * @internal
+ */
+export function UseThousandsSeparator(props: UseThousandsSeparatorProps) {
+  const { formatProps, onChange } = props;
+  const { translate } = useTranslation();
+
+  const useThousandsId = React.useId();
+
+  const handleSetFormatProps = React.useCallback(
```

Comment:
Looks like this function can be inlined into `setFormatTrait`

Reply:
thanks, made the change here and in other components that uses setFormatTrait